samza-core/src/main/java/org/apache/samza/application/ApplicationUtil.java

```diff
@@ -32,17 +34,37 @@
 public class ApplicationUtil {
 
   /**
-   * Creates the {@link SamzaApplication} object from the task or application class name specified in {@code config}
+   * Creates the {@link SamzaApplication} object from the task or application class name specified in {@code config}.
+   *
+   * This will use the current classloader to load the {@link SamzaApplication}.
    *
    * @param config the configuration of the application
    * @return the {@link SamzaApplication} object
    */
   public static SamzaApplication fromConfig(Config config) {
+    return fromConfig(config, false);
+  }
+
+  /**
+   * Creates the {@link SamzaApplication} object from the task or application class name specified in {@code config}.
+   * If {@code canApplyFrameworkIsolationIfEnabled} is true and isolation is enabled (in the config), then this will
+   * build the {@link SamzaApplication} using a special isolating classloader. Otherwise, this will use the current
```

Comment:
why do we need two levels (this boolean & the config) to determine the behavior? Can we not infer from the config and then use the appropriate loader?

Reply:
Currently, we only support using this classloader on the job coordinator. So we can't use this new flow on the processing containers nor `RemoteApplicationRunner` yet. Once we support split deployment on the processing containers and remove the need from `RemoteApplicationRunner`, then we can remove this flag. I will update the doc with that clarification.